File: HealthEase/HealthEase.Services/Database/Message.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace HealthEase.Services.Database
{
    public partial class Message : BaseEntity
    {
        public int MessageId { get; set; }
        public int PatientId { get; set; }
        public int UserId { get; set; }

        public int SenderId { get; set; } 
        public string SenderType { get; set; } 
        public string Content { get; set; } = "";
        public DateTime SentAt { get; set; } = DateTime.Now;
        public bool IsRead { get; set; } = false;

        public virtual Patient? Patient { get; set; }
        public virtual User? User { get; set; }
    }
}

File: HealthEase/UI/healthease_desktop/lib/models/message.dart
import 'package:healthease_desktop/models/patient.dart';
import 'package:healthease_desktop/models/user.dart';
import 'package:json_annotation/json_annotation.dart';
part 'message.g.dart';

@JsonSerializable()
class Message {
  int? messageId;
  int? patientId;
  int? userId;
  int? senderId;
  String? content;
  String? sentAt;
  String? senderType;
  bool? isRead;
  bool? isDeleted;
  User? user;
  Patient? patient;

  Message();

  factory Message.fromJson(Map<String, dynamic> json) =>
      _$MessageFromJson(json);

  /// Connect the generated [_$PersonToJson] function to the `toJson` method.
  Map<String, dynamic> toJson() => _$MessageToJson(this);
}
